pennylane/math/quantum.py

```diff
@@ -585,11 +671,14 @@ def purity(state, indices, check_state=False, c_dtype="complex128"):
         # in the return value, such that these properties are transferred to the result
         # as well. Adding 2 instead of 1 because state[0] could be -1 in some cases, and
         # converting to a real number for jax grad compatibility.
-        return np.real((state[0] + 2.0) / (state[0] + 2.0))
+        if len(np.shape(state)) == 1:
+            return np.real((state[0] + 2.0) / (state[0] + 2.0))
+
+        return np.real((state[0][0] + 2.0) / (state[0][0] + 2.0))

```

Comment:
do we not need to worry about other batch dims?

Reply:
Good catch 😄. This should indeed be a 1D array of 1s.